contrib/cirrus/pr-should-link-jira

```diff
@@ -0,0 +1,75 @@
+#!/bin/bash
+#
+# Intended for use in CI, for -rhel branches: check git PR, barf if no jira links
+#
+
+ME=$(basename $0)
+
+set -e
+
+# Github label which allows overriding this check
+OVERRIDE_LABEL="No Jira Link"
+
+# Only -rhel branches need jira links
+BRANCH_REGEX="v.*-rhel"
+
+LINK_REGEX="Fixes:? [\\t ]*https:\\/\\/issues.redhat.com\\/[\w\\/-]*"
+
+if [[ ! "${DEST_BRANCH}" =~ $BRANCH_REGEX ]]; then
+    exit 0
+fi
+
+if [[ "${CIRRUS_CHANGE_MESSAGE}" =~ $LINK_REGEX ]]; then
+    exit 0
+fi
+
+# Nope. Only allow if the github 'No Jira Link' label is set
+if [[ -z "$CIRRUS_PR" ]]; then
```

Comment:
This variable will be empty for executions on branches and tags, causing this script to fail and needlessly fire off monitoring alert e-mails.  I think maybe an easy fix is to simply set it to `exit 0` instead?

Reply:
This test is called from _run_validate-source, which is only run during PRs,, as per:
```
validate-source_task:
    name: "Validate source code changes"
    alias: validate-source
    # This task is primarily intended to catch human-errors early on, in a
    # PR context.  Skip running it everywhere else.
    only_if: &is_pr "$CIRRUS_PR != ''"
```
So, this is not a problem. 
In fact, this code is an identical copy from the similar pr-should-include-tests that already works for the same reason.

Reply:
You are correct, though scripts and tasks do get shuffled around now and again.  Would you consider:

```bash
if [[ -z "$CIRRUS_PR" ]]; then
    ...
elif [[ ! "$CIRRUS_BRANCH" =~ pull ]] || [[ -n "$CIRRUS_TAG" ]]; then
    echo "Warning: $ME only intended for use on PRs"
    exit 0
fi
```